client/gui/FramerateManager.cpp

```diff
@@ -11,12 +11,27 @@
 #include "StdInc.h"
 #include "FramerateManager.h"
 
+#include "../../lib/CConfigHandler.h"
+#include <SDL_video.h>
+
 FramerateManager::FramerateManager(int targetFrameRate)
-	: targetFrameTime(Duration(boost::chrono::seconds(1)) / targetFrameRate)
-	, lastFrameIndex(0)
+	: lastFrameIndex(0)
 	, lastFrameTimes({})
-	, lastTimePoint (Clock::now())
+	, lastTimePoint(Clock::now())
+	, vsyncEnabled(settings["video"]["vsync"].Bool())
 {
+	if(vsyncEnabled)
+	{
+		static int display_in_use = 0;
```

Comment:
Can you use settings["video"]["displayIndex"] instead of hardcoding the display?